### ISSUE TITLE
seminar_TRPO_tensorflow wrong formulae?

### ISSUE THREAD

**Issue by Pastafarianist** [yandexdataschool/Practical_RL#203]
The line

```
# KL divergence where first arg is fixed
```

Has the following expression

```
(tf.stop_gradient(tf.log(probs_out)) - tf.log(probs_out))
```

which is a constant 0. Is this a bug?

**Comment by Fritz449** [comment]
https://github.com/yandexdataschool/Practical_RL/commit/7a379ccfda5c23b3929b52e81cb6d2c91d13b2f8 - Fixed

**Comment by Fritz449** [comment]
This is not a mistake.

This expression is used to obtain the second derivatives of KL divergence at the start point. KL divergence at the start point is equal to zero (you can see this from the formula), but we propagate gradients through the second tf.log(probs_out).
